### PR TITLE
Update to Airship 14 Beta

### PR DESCRIPTION
**Pull request by BrandonStalnaker**

### CODE REVIEW

File: Example/mParticle-UrbanAirship.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -397,7 +397,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.3;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (rlepinski) [inline Example/mParticle-UrbanAirship.xcodeproj/project.pbxproj:400]: Airship still supports 11+, not sure if mParticle supports newer versions.

File: mParticle-UrbanAirship/MPKitUrbanAirship.m [hunk 2]
```diff
@@ -97,7 +97,7 @@ + (void)load {
 }
 
 + (NSSet *)defaultCategories {
-    return [UANotificationCategories createCategoriesFromFile:[[UAirship resources] pathForResource:@"UANotificationCategories" ofType:@"plist"]];
+    return [UANotificationCategories createCategoriesFromFile:[[UAirshipCoreResources bundle] resourcePath]];
```
Comment (rlepinski) [inline mParticle-UrbanAirship/MPKitUrbanAirship.m:100]: I dont think this change will work, I think you still need to do pathForResource:ofType:. Ill see if we can just expose a defaultCategories method

Comment (BrandonStalnaker) [reply mParticle-UrbanAirship/MPKitUrbanAirship.m:100]: pathForResource:ofType: doesn't seem to be public

Comment (rlepinski) [reply mParticle-UrbanAirship/MPKitUrbanAirship.m:100]: We added `[UANotificationCategories defaultCategories]` - https://docs.airship.com/reference/libraries/ios/latest/Airship/Classes/UANotificationCategories.html#/c:objc(cs)UANotificationCategories(cm)defaultCategories

**Comment by rlepinski** [comment]
After the `defaultCategories` change, the only other thing I can see is maybe the import. We use this in our react module to handle both `use_frameworks!` and not using frameworks:

```
#if __has_include("AirshipLib.h")
#import "AirshipLib.h"
#else
@import Airship;
#endif
```

**Review by rlepinski — APPROVED** [review]

**Pull request merged**